recipes/gsoap/all/conanfile.py

```diff
@@ -63,7 +63,7 @@ def _configure_cmake(self):
             return self._cmake
         self._cmake = CMake(self)
         self._cmake.definitions["GSOAP_PATH"] = self._source_subfolder
-        self._cmake.definitions["BUILD_TOOLS"] = True
+        self._cmake.definitions["BUILD_TOOLS"] = not tools.cross_building(self.settings)
```

Comment:
I'm not a fan of creating different packages (same hash, different content) depending on the build system.
I would add a `tools` option and fail in `configure` when cross building and this option is set.

Reply:
Agree with this comment. Every change in the resulting package should be a consequence o changing an option or a setting value. In this case, why not leaving the build tools def to `True`?

Reply:
Build tools true when cross compiling fails on itself due to recursion.

Reply:
I just noticed that we don't use the gsoap autotools build system at all.
We apparently have forked our own cmake system.
See https://github.com/conan-io/conan-center-index/pull/1699 for the pr adding gsoap. I don't know what problems the autotools build sytem has? (@ericLemanissier  @madduci)

Anyhow:
- the cmake system has not been used to cross build between Windows and Linux due to it wanting to use winflexbison on mingw@Linux.
- the test_package does not work when the tools aren't built

I've created https://github.com/conan-io/conan-center-index/pull/3615 which addresses some of these issues.

@KristianJerpetjon 
Can you try it? I've created a new `tools` option.

Reply:
I have no idea why upstream autotools system is not used